src/librbd/api/Mirror.cc

```diff
@@ -3190,12 +3189,37 @@ int Mirror<I>::group_resync(IoCtx& group_ioctx, const char *group_name) {
   }
 
   if (state == cls::rbd::MIRROR_SNAPSHOT_STATE_PRIMARY) {
-    lderr(cct) << "group " << group_name
+    lderr(cct) << "group=" << group_name
                << " is primary, cannot resync to itself" << dendl;
     return -EINVAL;
   }
 
-  // TODO: implement the group resync functionality
+  r = cls_client::mirror_group_resync_set(&group_ioctx,
```

Comment:
Instead of creating a new object for this, it is better to add a meta attribute to the rbd group header object.
This prevents data duplication and potential issues with managing the new rados object.